lib/ddtrace/configuration/components.rb

```diff
@@ -55,7 +56,14 @@ def build_tracer(settings, agent_settings)
             default_service: settings.service,
             enabled: settings.tracer.enabled,
             partial_flush: settings.tracer.partial_flush.enabled,
-            tags: build_tracer_tags(settings)
+            tags: build_tracer_tags(settings),
+            sampler: PrioritySampler.new(
+              base_sampler: AllSampler.new,
+              post_sampler: Sampling::RuleSampler.new(
+                rate_limit: settings.sampling.rate_limit,
+                default_sample_rate: settings.sampling.default_rate,
+              )
+            )
```

Comment:
This is actually the default behaviour, it's kind of hard to reason about this today, because whatever sampler you pass to `sampler:` it gets converted to a `PrioritySampler` if not already.
The default is an `AllSampler` as `base_sampler` and `RuleSampler` as `post_sampler`..

Btw, this changes is here because letting `Datadog::Tracer` resolve the default sampler puts us in a circular dependency between tracer initialization and configuration initialization.